framework/helpers/BaseStringHelper.php

```diff
@@ -268,4 +268,37 @@ public static function explode($string, $delimiter = ',', $trim = true, $skipEmp
         }
         return $result;
     }
+
+    /**
+     * Explodes string into array using regular expression, optionally trims values and skips empty ones
+     *
+     * @param string $string String to be exploded.
+     * @param string $pattern Delimiter regular expression. Default is `/[\s,;]+/`.
+     * @param mixed $trim Whether to trim each element. Can be:
+     *   - boolean - to trim normally;
+     *   - string - custom characters to trim. Will be passed as a second argument to `trim()` function.
+     *   - callable - will be called for each value instead of trim. Takes the only argument - value.
+     * @param boolean $skipEmpty Whether to skip empty strings between delimiters. Default is false.
+     * @return array
+     * @since 2.0.7
+     */
+    public static function pregExplode($string, $pattern = '/[\s,;]+/', $trim = true, $skipEmpty = false)
+    {
+        $result = preg_split($pattern, $string);
+        if ($trim) {
+            if ($trim === true) {
+                $trim = 'trim';
+            } elseif (!is_callable($trim)) {
+                $trim = function ($v) use ($trim) {
+                    return trim($v, $trim);
+                };
+            }
+            $result = array_map($trim, $result);
+        }
+        if ($skipEmpty) {
+            // Wrapped with array_values to make array keys sequential after empty values removing
+            $result = array_values(array_filter($result));
```

Comment:
array_filter will also remove string `0`, I doubt that's expected? https://3v4l.org/dXpPr